app/models/ending_time.rb

```diff
@@ -0,0 +1,21 @@
+class EndingTime
+  def initialize(params)
+    @ending_time_params = params.select { |key, _| key.include?("ending_time") }
+  end
+
+  def build
+    if ending_time.empty? || ending_time == ":"
+      nil
+    else
+      Time.parse(ending_time).to_s
+    end
+  end
+
+  private
+
+  attr_reader :ending_time_params
+
+  def ending_time
+    ending_time_params.values.last(2).join(":").to_s
+  end
+end
```

Comment:
Actually, on second thought, since the result differs between them - we return `nil` for a missing ending time and an empty string for missing starting time, I do not think it is worth the extra complexity to handle these cases in a single object.